pkg/api/norman/customization/aks/listers.go

```diff
@@ -178,6 +192,34 @@ func listVirtualNetworks(ctx context.Context, cap *Capabilities) ([]byte, int, e
 	return encodeOutput(networks)
 }
 
+type locationsResponseBody struct {
+	Name        string `json:"name"`
+	DisplayName string `json:"displayName"`
+}
+
+func listLocations(ctx context.Context, cap *Capabilities) ([]byte, int, error) {
+	clientSubscription, err := NewSubscriptionServiceClient(cap)
+	if err != nil {
+		return nil, http.StatusInternalServerError, err
+	}
+
+	locationList, err := clientSubscription.ListLocations(ctx, cap.SubscriptionID)
+	if err != nil {
+		return nil, http.StatusBadRequest, fmt.Errorf("failed to get locations: %v", err)
+	}
+
+	var locations []locationsResponseBody
+
+	for _, location := range *locationList.Value {
+		locations = append(locations, locationsResponseBody{
+			Name:        to.String(location.Name),
+			DisplayName: to.String(location.DisplayName),
+		})
+	}
+
+	return encodeOutput(locations)
+}
+
 func encodeOutput(result interface{}) ([]byte, int, error) {
 	data, err := json.Marshal(&result)
 	if err != nil {
```

Comment:
Can you also address this: https://github.com/rancher/rancher/pull/32656/files#r639039462